addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,337 @@
+                } else if ("power_off".equals(command)) {
+                    power = false;
+                    updateState(getThing().getChannel("mute").getUID(), OnOffType.OFF);
+                    updateState(getThing().getChannel("power").getUID(), OnOffType.OFF);
+                    updateState(getThing().getChannel("volume").getUID(), UnDefType.NULL);
+                    updateState(getThing().getChannel("source").getUID(), UnDefType.NULL);
+                } else if ("power_on".equals(command)) {
+                    power = true;
+                    updateState(getThing().getChannel("power").getUID(), OnOffType.ON);
+                    powerOnRefresh();
+                } else if ("power".equals(command)) {
+                    String state = readUntil('!');
+                    if ("on".equals(state)) {
+                        power = true;
+                        updateState(getThing().getChannel("power").getUID(), OnOffType.ON);
+                        powerOnRefresh();
+                    } else if ("standby".equals(state)) {
+                        updateState(getThing().getChannel("mute").getUID(), OnOffType.OFF);
+                        updateState(getThing().getChannel("power").getUID(), OnOffType.OFF);
+                        power = false;
+                    }
+                } else if ("dimmer".equals(command)) {
+                    updateState(getThing().getChannel("dimmer").getUID(), readDimmer());
+                } else if ("freq".equals(command)) {
+                    updateState(getThing().getChannel("frequency").getUID(), readFrequency());
+                } else if ("source".equals(command)) {
+                    updateState(getThing().getChannel("source").getUID(), new StringType(readUntil('!')));
+                } else if ("display".equals(command)) {
+                    String stringLength = readUntil(',');
+                    int length = Integer.parseInt(stringLength);
+                    byte[] data = new byte[length];
+                    int off = 0;
+                    while (off != length) {
+                        int r = serialPort.getInputStream().read(data, off, length - off);
+                        if (r == -1) {
+                            throw new IOException("Connection closed while reading display content");
+                        } else {
+                            off += r;
+                        }
+                    }
+                    // We don't do anything with display content, could add a channel if this is useful
+                } else {
+                    readUntil('!'); // discard
+                }
+
+            } catch (IOException e) {
+                if (serialPort != null) {
+                    logger.error("Input error while receiving data from amplifier", e);
+                    disconnect();
+                }
+            } catch (NullPointerException e) {
```

Comment:
You should not catch `NullPointerException` but should prevent it from happening, can you resolve the underlying problem?